### PR TITLE
Git Article init

### PR DESCRIPTION
**Pull request by BCNelson**
Feel free to push changes to the article-get branch and in general discuss the article

### CODE REVIEW

**Comment by WilhelmMcStretch** [comment]
article additions for contribution actions like pull requests, forks, merges, clone, etc

File: docs/tools/git.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Branches
+
+Branches are used for having multiple versions of the files with diverging timelines. Lets say, for a moment, that our fictional author wants to add a twist to his story. It is going to be a complicated change requiring him to go back and edit much of what he has already written and he is not sure if he is going to want to keep this twist in. If we are using Git then the most logical thing to do is to create a branch in the repo; much like a copy the writer can make all of his changes and then decide if he wants to keep them. This use case alone does not warrant the use of these branching timelines[^2]. We see the real power of using branches when we start having multiple authors. One author can then be working on the twist in the story and the other can be working another part of the story. They can then merge the changes that both made into the final product.
+
+[^2]: This particular problem could be solved by simply making a copy of the book and then decide which copy you want to go with.
```
Comment (py-crash) [inline docs/tools/git.md:19]: This reference [^2] doesn't work in our mkdocs version

File: docs/tools/git.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Git
+
+Git is version control software and is used to allow for collaborative projects.
```
Comment (py-crash) [inline docs/tools/git.md:3]: Mention that is used in this wiki.

File: docs/tools/git.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+[^2]: This particular problem could be solved by simply making a copy of the book and then decide which copy you want to go with.
+
+### Example
+
+Alice and Bob are writing a story together. Alice decides that they should have the mentor of the main character turn out to be a close relive. They want it to be hinted at throughout the book, but that requires changes throughout the book. They are not sure if the story is going to turn out well that way and it will be hard to separate out the changes that Alice makes if they both make them on the same version of the book. This is because there is no point in history of the book that all of Bob's changes will be made but none of Alice's. The solution is to use branches to keep track of the changes separately and then merge the changes if desired.
+
+### Practical
+
+By convention the main branch of a repo is called the `master` branch. It holds the latest accepted code. There are several other common branches `develop` is the other most common. You can check out the most current commit of a branch with `git checkout <branch-name>`
```
Comment (py-crash) [inline docs/tools/git.md:27]: Would it be a good idea to say that there may also be branches for different features?

File: docs/tools/git.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## Commits
+
+Git tracks changes to text based files and wraps those changes into packages called commits. These commits are the basic building blocks of a git repository. A branch is a list of commits.
```
Comment (py-crash) [inline docs/tools/git.md:31]: I would add:

> In each commit the changes made to the repo are explained by the autor.

**Pull request merged**